### PR TITLE
feat!: use current inquirer

### PR DESCRIPTION
**Pull request by mshanemc**
BREAKING CHANGE: ESM-only

BREAKING CHANGE: removes cross-cloud hooks (deployer, deauthorizer, etc)

BREAKING CHANGE: removes Prompter module

BREAKING CHANGE: remove option to barrel-import from Ux

@W-14674359@

---

The following PRs use the pre-release package as "proof of correctness" and should be updated once this merges.  They can be checked out and built to do manual QA.  They also explore what changes this PR will cause.

- https://github.com/salesforcecli/plugin-auth/pull/902
- https://github.com/salesforcecli/plugin-signups/pull/473
- https://github.com/salesforcecli/plugin-org/pull/917
- https://github.com/salesforcecli/plugin-settings/pull/484
- https://github.com/salesforcecli/plugin-trust/pull/697
- https://github.com/salesforcecli/plugin-packaging/pull/537 (see note about changes for `this.project`)
- https://github.com/salesforcecli/sf-plugins-core/pulls

These PRs aren't meant to require the new sf-plugins-core

- https://github.com/salesforcecli/plugin-sobject/pull/489

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,9 +17,7 @@ The SfCommand abstract class extends [@oclif/core's Command class](https://githu
 - Enables additional help sections to the standard oclif command help output
 - Provides access to the [cli-ux cli actions](https://github.com/oclif/cli-ux#cliaction). This avoids having to import that interface from cli-ux and manually handling the `--json` flag.
 
-## Prompter Class
-
-A general purpose class that prompts a user for information. See [inquirer NPM Module](https://www.npmjs.com/package/inquirer) for more information.
+## Prompts
```
Comment (mdonnalley) [inline README.md:20]: Do you want to add some content here?

**Review by mdonnalley — APPROVED** [review]
Changes look good to me 👍 

Looks like you have some merge conflicts though. Also, I'm assuming that the failing nuts are expected?

**Comment by mdonnalley** [comment]
### QA

Generated new sf plugin using `sf dev generate plugin`, linked this branch of sf-plugins-core to it.

🟢 `this.confirm` prompts user for `Y/n` answer
🟢 `this.confirm` defaults to `defaultAnswer` after 10 second timeout
❌  `this.confirm` hangs until timeout despite being answered

🟢  `this.secretPrompt` prompts user and input is hidden
❌  `this.secretPrompt` hangs until timeout despite being answered

**Comment by mdonnalley** [comment]
QA
Generated new sf plugin using sf dev generate plugin, linked this branch of sf-plugins-core to it.

🟢 `this.confirm` prompts user for Y/n answer
🟢 `this.confirm` defaults to defaultAnswer after 10 second timeout
🟢 `this.confirm` cancels timeout if answer is given

🟢 `this.secretPrompt` prompts user and input is hidden
🟢 `this.secretPrompt` cancels timeout if answer is given

**Comment by mshanemc** [comment]
> Changes look good to me 👍
> 
> Looks like you have some merge conflicts though. Also, I'm assuming that the failing nuts are expected?

PDR needs https://github.com/salesforcecli/plugin-deploy-retrieve/pull/866 to even compile without the `Deployer` hook stuff.

The other 2 (org and signups) are expected results of the breaking changes in here.

File: MIGRATING_V5-V6.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+v6 is ESM-only, which can't be consumed by plugins written in cjs. Salesforce-owned plugins are ESM now and you can use them as a guide for the necessary changes.
+
+## Prompts
+
+This library uses [`inquirer`](https://github.com/SBoudrias/Inquirer.js) for interactivity. Inquirer made some large changes, including its own ESM rewrite. To take advantage of its improved performance and smaller dependency, we've upgraded.
+
+The API is completely different, resulting in changes to sf-plugins-core. The new philopsophy is
+
+1. provide limited, simplified prompts for common use cases
+1. plugins that need more advanced propting should import the parts of inquirer that they need
```
Comment (mdonnalley) [inline MIGRATING_V5-V6.md:14]: `2.`

File: MIGRATING_V5-V6.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### Changes
+
+The `Prompter`` class is removed.
```
Comment (mdonnalley) [inline MIGRATING_V5-V6.md:18]: There's an extra tick at the end of `Prompter`

File: MIGRATING_V5-V6.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+SfCommand contains two prompt methods
+
+1. `confirm` provides boolean confirmation prompts
+1. `secretPrompt` takes masked string input from the user
```
Comment (mdonnalley) [inline MIGRATING_V5-V6.md:23]: `2.`

File: MIGRATING_V5-V6.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Unlike then inquirer base prompts (`confirm` and `password`, respectively) these have a built-in default timeout. Both take an object parameter that lets you change the timeout (confirm previously took a series of parameters)
```
Comment (mdonnalley) [inline MIGRATING_V5-V6.md:25]: `Unlike the`

**Review by mdonnalley — CHANGES_REQUESTED** [review]

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**